FrontBundle/SubQuery/Strategies/DeviceSubQueryStrategy.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace OpenOrchestra\FrontBundle\SubQuery\Strategies;
+
+
+/**
+ * Class DeviceSubQueryStrategy
+ */
+class DeviceSubQueryStrategy extends AbstractRequestSubQueryStrategy
+{
+    /**
+     * @param string $blockParameter
+     *
+     * @return bool
+     */
+    public function support($blockParameter)
+    {
+        return 'device' === $blockParameter;
+    }
+
+    /**
+     * @param string $blockParameter
+     *
+     * @return array
+     */
+    public function generate($blockParameter)
+    {
+        if ($device = $this->request->headers->get('x-ua-device')) {
+            return array('x-ua-device' => $device);
```

Comment:
As 'x-ua-device' string appears at different places, it could be a good idea to make it a constant